### PR TITLE
Do not copy /etc/resolv.conf if empty

### LINKED ISSUE #185: More robust use of name resolution setup
The DMS uses the hosts resolv.conf file to implement the DNS setup. However depending on the host configuration the resolv.conf information is cleared out on reboot of the machine. In this case the DMS system should not copy the empty resolv.conf information but rather stick with the DHCP data the DMS received at boot time.

### PR DESCRIPTION
**Pull request by jesusbv**
This Fixes #185
This references bsc#1182234

### CODE REVIEW

File: suse_migration_services/units/setup_host_network.py [hunk 1]
```diff
@@ -139,3 +142,18 @@ def log_network_details():
...
+        lines = filter(
+            lambda line: not line.startswith('#') and not line.startswith(os.linesep),
+            resolv.readlines()
+        )
+    for line in lines:
+        # check there is useful information in the remaining lines
+        if line.startswith('search') or line.startswith('nameserver'):
+            return False
+
+    return True
```
Comment (schaefi) [inline suse_migration_services/units/setup_host_network.py:159]: Maybe we could write this a bit simpler ? like

```python
with open(resolv_conf_path, 'r') as resolv:
    for line in resolv:
        if line.startswith('search') or line.startswith('nameserver'):
            # looks like there is some effective resolver setup
            return False
return True
```

maybe it's also easier to read if the function is named "def have_host_resolv_setup" or similar

Thoughts ?

File: test/unit/units/setup_host_network_test.py [hunk 2]
```diff
@@ -91,3 +98,80 @@ def test_main(
...
+
+    @patch('suse_migration_services.units.setup_host_network.is_resolv_empty')
+    @patch('suse_migration_services.logger.Logger.setup')
+    @patch('suse_migration_services.command.Command.run')
+    @patch('suse_migration_services.units.setup_host_network.Fstab')
+    @patch('os.path.exists')
+    @patch('shutil.copy')
+    @patch('glob.glob')
+    @patch('os.path.isfile')
+    def test_main_no_copy(
```
Comment (schaefi) [inline test/unit/units/setup_host_network_test.py:110]: I think the intention of this test is to check that etc/resolv conf is not copied. I wonder about all the other assertions for fstab and so on

Comment (schaefi) [reply test/unit/units/setup_host_network_test.py:110]: I would have expected something like ```assert not mock_shutil_copy.called``` or ```assert call... not in mock_shutil_copy.call_args_list```. Something that is more explicit to what the test is good for. Does this makes sense ?

**Review by schaefi — CHANGES_REQUESTED** [review]
Thanks for looking into this. I think the logic makes good sense, but I think it can be reached with less code. Also think about unit tests as code, less code is better and easier to read, especially in tests. Hope that makes sense ?

File: test/unit/units/setup_host_network_test.py (hunk 2 above)
Comment (schaefi) [reply test/unit/units/setup_host_network_test.py:110]: All the other conditions of main are part of other test cases. So I think it's fine if this test just focuses on no copy happened in the most easy way to get that condition

Comment (jesusbv) [reply test/unit/units/setup_host_network_test.py]: Fixed.

File: suse_migration_services/units/setup_host_network.py (hunk 1 above)
Comment (jesusbv) [reply suse_migration_services/units/setup_host_network.py]: Done.

**Review by schaefi — APPROVED** [review]
yeah, that looks really nice now :+1:

**Pull request merged**